Cleaner and More Robust 'conf' File

terminalforlife opened the pull request:
This _should_ be a considerable improvement, but I urge testing beforehand. Most of the changes are on `confget()`, for now. Please let me know if there are any issues.

Commit:
Replace 'ls' with '['

Typically, you never want to do something like what was there prior to
this commit. There are a number of file-related test expressions I
recommend looking into.

See `man bash`, for more information, then look for the section titled
"CONDITIONAL EXPRESSIONS", which lists the available expressions.

Commit:
Merge branch 'master' of https://github.com/paperbenni/bash

Commit:
Rework confget()

Following an E-Mail exchange, I've written and tested this in various
ways, on my end, without any issues.

The behavior of the function should be the same as the previous code.
If it's expected to have thousands of lines of keys and values to parse
in the configuration file, then I suggest _not_ merging this commit.

Commit:
Use extglob in confget() to remove leading spaces

See `man bash` then search for 'extglob', for more information. It's a
shell option (accessible via the `shopt` bash builtin) allowing for
advanced glob matching; this is the next best thing to REGEX.

I recall your original code allowed for removing a leading space, so
now that feature is back again, only, any leading spaces are moved.
Even tabs.

If you never plan to have spaces of any kind of your 'value' part of
the key:value system, then you could omit the extglob usage and instead
use: `${ARRAY[1]// }` when echoing out the value of the searched key.
This will just substitute all (global) spaces for nothing, thus
removing them.

I've reinforced the both functions by ensuring the file exists and has
the correct permissions. It'll just help avoid any ugly errors.

Commit:
Uncomment line not used while testing

ghost commented:
Regarding `read` and CodeFactor saying it's an "issue": why one would want to _interpret_ escape sequences in a configuration file is beyond me, right now. I'd easily argue it'd be rarer to need to actually interpret escape sequences.

Commit:
Address CodeFactor issue

Apparently `-f` doesn't work with globs. From testing, I got the
desired results, so either my tests were terrible or it's a case of the
site misunderstanding the code.

Commit:
Address CodeFactor issue with `-a` and `-o`

After thinking about it, I realised it's possible albeit improbable, -
that a configuration file's name could contain strange symbols which
would interfere with the test.

ghost commented:
I had a brain-fart, regarding the `-f` and glob in the test. It worked when I tried it because I only had one file; of course, had I multiple files it would have expanded as one would normally expect, which would break the test. Oops.
